evalml/tests/pipeline_tests/test_component_graph.py

```diff
@@ -836,3 +842,191 @@ def test_component_graph_sampler_same_given_components(mock_dt_fit, mock_rf_fit)
     component_graph2.fit(X, y)
     assert mock_dt_fit.call_args[0][0] == mock_rf_fit.call_args[0][0]
     assert mock_dt_fit.call_args[0][1] == mock_rf_fit.call_args[0][1]
+
+
+class LogTransform(TargetTransformer):
+    name = "Log Transform"
+
+    def __init__(self, parameters=None, random_seed=0):
+        super().__init__(parameters={}, component_obj=None, random_seed=random_seed)
+
+    def fit(self, X, y):
+        return self
+
+    def transform(self, X, y=None):
+        if y is None:
+            return X, y
+        y = infer_feature_types(y)
+        return X, infer_feature_types(np.log(y.to_series()))
+
+    def inverse_transform(self, y):
+        y = infer_feature_types(y)
+        return infer_feature_types(np.exp(y.to_series()))
+
+
+class DoubleTransform(TargetTransformer):
+    name = "Double Transform"
+
+    def __init__(self, parameters=None, random_seed=0):
+        super().__init__(parameters={}, component_obj=None, random_seed=random_seed)
+
+    def fit(self, X, y):
+        return self
+
+    def transform(self, X, y=None):
+        if y is None:
+            return X, y
+        y = infer_feature_types(y).to_series()
+        return X, infer_feature_types(y * 2)
+
+    def inverse_transform(self, y):
+        y = infer_feature_types(y).to_series()
+        return infer_feature_types(y / 2)
+
+
+class SubsetData(Transformer):
```

Comment:
To simulate a transformer that modifies the target but is not a target transformer, e.g. a sampler